src/libraries/System.Windows.Extensions/src/System/Media/SoundPlayer.cs

```diff
@@ -508,23 +509,22 @@ private async Task CopyStreamAsync(CancellationToken cancellationToken)
                     readBytes = await _stream.ReadAsync(_streamData.AsMemory(_currentPos, BlockSize), cancellationToken).ConfigureAwait(false);
                     totalBytes += readBytes;
                 }
-
-                _lastLoadException = null;
             }
-            catch (Exception exception)
+            catch (Exception ex)
             {
-                _lastLoadException = exception;
+                exception = ex;
             }
 
+            _lastLoadException = exception;
```

Comment:
I'm not sure it's necessary to set _lastLoadException here unless there actually was an exception. However to minimise the change, it still sets null as it did before.